recipes/tests.rb

```diff
@@ -381,6 +391,25 @@ module load intelmpi && mpirun --help | grep 'Version 2019 Update 8'
   TESTCUDA
 end
 
+###################
+# FabricManager
+###################
+bash 'test fabric-manager daemon' do
+  cwd Chef::Config[:file_cache_path]
+  code <<-TESTFM
+    nvswitches=$(lspci -d 10de:1af1 | wc -l)
+
+    if [ ${nvswitches} -lt 1 ]; then
+      echo "No nvswitches detected, no test needed."
+      exit 0
+    fi
+
+    set -e
+    systemctl status nvidia-fabricmanager.service | grep "Active: active (running)"
```

Comment:
There is a better way the to check the status of `systemctl` service. This because `systemctl status` also shows some line of logs that can inadvertently false the grep check.
Use `systemctl show -p SubState nvidia-fabricmanager | grep -i running`